File: gamehub-server/src/main/java/com/gamehub/api/model/user/UserResponseModel.java
package com.gamehub.api.model.user;

import com.gamehub.persistance.entity.UserStatus;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
public class UserResponseModel {
    private Long id;
    private String username;
    private String email;
    private UserStatus status;
    private String fullName;
    private String bio;
    private String profilePictureUrl;
    private LocalDateTime lastLogin;
    private Set<String> roles;
}

File: gamehub-android/app/src/main/java/com/gamehub/model/user/UserResponseModel.kt
package com.gamehub.model.user

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

//@Parcelize
data class UserResponseModel(
    val id: Long,
    val username: String,
    val email: String,
    val status: String,
    val fullName: String,
    val bio: String,
    val profilePictureUrl: String,
    val lastLogin: String,
    val roles: Set<String>
)
